### PR TITLE
Implementation of git xet cp, upload only

### PR DESCRIPTION
**Pull request by hoytak**
This PR implements git xet cp, which should follow the xet cp behavior nearly exactly.  

Currently, it only implements local-to-xet uploads currently, with local-to-local also enabled for tests.  xet-to-local and xet-to-xet would be the next step and would be doable.

This PR: 
- Adds a file_operations/ mod inside xetblob for handling batched file operations.
- Imports the rust transaction tools in pyxet that limit the size and number of in-flight transactions and files per transaction.
- Ports the perform_copy function in pyxet into rust. 
- Adds git-xet cp as the cli command for running this.
- Adds tests of the perform_copy function that operate on local-to-local copies.

### CODE REVIEW

File: rust/gitxetcore/src/xetblob/file_operations/batched_wfile_handle.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    /// Loads data from a file into this handle to the remote file writer.
+    pub async fn upload_from_file_and_close(mut self, path: impl AsRef<Path>) -> Result<()> {
+        let mut file = std::fs::File::open(path)?;
+
+        let mut buffer = vec![0u8; WRITE_FILE_BLOCK_SIZE];
+
+        loop {
+            let n_bytes = file.read(&mut buffer[..])?;
```
Comment (ylow) [inline rust/gitxetcore/src/xetblob/file_operations/batched_wfile_handle.rs:45]: check for ErrorKind::Interrupted and resume.

File: rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+use std::sync::Arc;
+
+use crate::errors::{GitXetRepoError, Result};
+use crate::git_integration::git_url::parse_remote_url;
+use crate::xetblob::XetRepoManager;
+
+struct CPOperation {
+    src_path: String,
+    dest_dir: String,
+    dest_path: String,
```
Comment (ylow) [inline rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs:15]: Whats in dest_dir and dest_path?  is it meant to be dest_dir and dest_filename? But looks like dest_path could contain slashes too?

File: rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+                        },
+                    ));
+                }
+            } else {
+                return Err(GitXetRepoError::InvalidOperation(format!(
+                    "Copy: source {src} is a directory, but cp not recursive."
+                )));
+            }
+        } else {
+            cp_ops.push(CPOperation {
```
Comment (ylow) [inline rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs:169]: If src is file and dest is directory do you need to take the src filename and append to dest?  (I am still not sure what is dest_dir and dest_path)

File: rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs (hunk 2 above)
Comment (hoytak) [reply rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs:16]: Added comments.

File: rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs (hunk 3 above)
Comment (hoytak) [reply rust/gitxetcore/src/xetblob/file_operations/cp_operations.rs:171]: That is done above.

File: rust/gitxetcore/src/xetblob/file_operations/batched_wfile_handle.rs (hunk 1 above)
Comment (hoytak) [reply rust/gitxetcore/src/xetblob/file_operations/batched_wfile_handle.rs]: Done!  Good thought.

**Review by ylow — APPROVED** [review]

**Pull request merged**